core.go

```diff
@@ -1478,6 +1478,47 @@ func MinMaxLoc(input Mat) (minVal, maxVal float32, minLoc, maxLoc image.Point) {
 	return float32(cMinVal), float32(cMaxVal), minLoc, maxLoc
 }
 
+// Copies specified channels from input arrays to the specified channels of output arrays.
+//
+// For further details, please see:
+// https://docs.opencv.org/master/d2/de8/group__core__array.html#ga51d768c270a1cdd3497255017c4504be
+//
+func MixChannels(src []Mat, dst []Mat, fromTo []int) {
+	cSrcArray := make([]C.Mat, len(src))
+	for i, r := range src {
+		cSrcArray[i] = r.p
+	}
+	cSrcMats := C.struct_Mats{
+		mats:   (*C.Mat)(&cSrcArray[0]),
+		length: C.int(len(src)),
+	}
+
+	cDstArray := make([]C.Mat, len(dst))
+	for i, r := range dst {
+		cDstArray[i] = r.p
+	}
+	cDstMats := C.struct_Mats{
+		mats:   (*C.Mat)(&cDstArray[0]),
+		length: C.int(len(dst)),
+	}
+
+	cFromToArray := make([]C.int, len(fromTo))
+	for i, ft := range fromTo {
+		cFromToArray[i] = C.int(ft)
+	}
+
+	cFromToIntVector := C.IntVector{
+		val:    (*C.int)(&cFromToArray[0]),
+		length: C.int(len(fromTo)),
+	}
+
```

Comment:
I added locally, accordingly all the defers close mats and close intvector:
````
defer C.Mats_Close(cSrcMats)
defer C.Mats_Close(cDstMats)
defer C.IntVector_Close(cFromToIntVector)
````
 But they are failing locally, for instance:
````
goroutine 0 [idle]:
runtime: unknown pc 0x7f1645a1c438
`````
If I remove the defer the test passes, but the [build fails](https://ci.appveyor.com/project/deadprogram/gocv/builds/35603805#L3469) because of `TestMatProfile`. Any cluewhat I am doing wrong?

Reply:
Well, at first glance you do not need any of the defers there. For example `defer C.Mats_Close(cDstMats)` would close the `dst` value, which is not what you would want right after returning.

I think perhaps the test itself it the issue. See comment below.